tiledb/bioimg/converters/ome_tiff.py

```diff
@@ -274,17 +286,86 @@ def __init__(self, output_path: str):
+        baseline: bool,
+        num_levels: int,
+        image: np.ndarray,
+        metadata: Mapping[str, Any],
     ) -> None:
-        write_kwargs = json.loads(metadata["json_write_kwargs"])
-        self._writer.write(image, **write_kwargs)
+        tiffwriter_metadata = json.loads(
+            metadata.get("writer_metadata", {}).get("json_write_kwargs", "{}")
+        )
+        axes_metadata = metadata.get("group_metadata", {}).get("axes", [])[level]
+        channel_metadata = (
+            metadata.get("group_metadata", {}).get("channels", {}).get(ATTR_NAME, [])
+        )
+        original_axes = "".join(axes_metadata.get("originalAxes"))
+
+        write_kwargs: Dict[str, Any] = {}
+        if baseline:
+            write_kwargs["subifds"] = num_levels - 1 if num_levels > 1 else None
+        else:
+            write_kwargs["subfiletype"] = tifffile.FILETYPE.REDUCEDIMAGE
+
+        write_kwargs["tile"] = (256, 256)
+        write_kwargs["metadata"] = {
+            "axes": original_axes,
+            "Channel": {
+                "Name": [
+                    channel.get("name", f"Channel {idx}")
+                    for idx, channel in enumerate(channel_metadata)
+                ],
+                "Color": [
+                    str(get_decimal_from_rgba(channel.get("color")))
+                    for channel in channel_metadata
+                ],
+            },
+        }
+
+        ctx = decimal.Context()
+        ctx.prec = 32
+
+        for dim in ["X", "Y", "Z"]:
+            if dim in original_axes:
+                write_kwargs["metadata"][f"PhysicalSize{dim}"] = format(
+                    ctx.create_decimal(repr(metadata.get(f"physicalSize{dim}", 1))), "f"
+                )
+                write_kwargs["metadata"][f"PhysicalSize{dim}Unit"] = metadata.get(
+                    f"physicalSize{dim}Unit", "µm"
+                )
+
+        if "T" in original_axes:
+            write_kwargs["metadata"]["TimeIncrement"] = format(
+                ctx.create_decimal(repr(metadata.get("timeIncrement", 1))), "f"
+            )
+            write_kwargs["metadata"]["TimeIncrementUnit"] = metadata.get(
+                "timeIncrementUnit", "s"
+            )
+
+        mask = []
+        for _ in original_axes:
+            mask.append(slice(None))
+
+        if len(channel_metadata) == 3 and image.dtype == np.dtype(np.uint8):
+            write_kwargs["compression"] = tifffile.COMPRESSION.JPEG
+            write_kwargs["compressionargs"] = {"level": 70}
+            write_kwargs["photometric"] = tiffwriter_metadata.get(
+                "photometric", tifffile.PHOTOMETRIC.RGB
+            )
+            mask[original_axes.index("C")] = slice(0, 3)  # Drop alpha channel
+        else:
+            write_kwargs["compression"] = tifffile.COMPRESSION.ADOBE_DEFLATE
+            write_kwargs["photometric"] = tiffwriter_metadata.get(
+                "photometric", tifffile.PHOTOMETRIC.MINISBLACK
+            )
```

Comment:
Why all this code inflation? Is it necessary to have group metadata in array metadata? Aren't the same for every level?